### DISCUSSION TITLE
102. MySQL의 스토리지 엔진 간 차이점이 무엇인가요?

### DISCUSSION THREAD

**Discussion by jongmee** [jewelry-box/cs-interview#106] [General]
1. MySQL의 스토리지 엔진 간 차이점이 무엇인가요?
2. 잠금에 대해서는 어떻게 다른가요?

키워드: MySQL 스토리지 엔진, MyISAM, InnoDB

**Comment by cutehumanS2** [comment]
MySQL의 스토리지 엔진으로는 대표적으로 MyISAM과 InnoDB가 있습니다.
이 두 스토리지 엔진의 주요 차이점은 **트랜잭션 지원 여부**와 **잠금 방식**입니다.

- **MyISAM**은 트랜잭션을 지원하지 않아 데이터의 ACID 보장이 어렵습니다. 그리고 데이터를 변경할 때 테이블 수준의 잠금을 사용하여 동시성 처리 성능이 낮습니다. 따라서 로그 저장이나 읽기 위주의 작업에 적합합니다.
- **InnoDB**는 ACID를 준수하는 트랜잭션을 지원하여 데이터 정합성을 보장합니다. 그리고 레코드 수준의 잠금을 사용하여 필요한 행만 잠그기 때문에, 동시성 처리 성능이 좋습니다. 따라서 은행 시스템이나 전자 상거래 같은 도메인에 적합합니다.
    - 추가적으로 외래 키 제약 조건과, 클러스터형 인덱스를 지원한다는 특징도 있습니다.

<br><br>

[참고한 자료]
- https://dev.mysql.com/doc/refman/8.0/en/innodb-introduction.html
- https://www.youtube.com/watch?v=IftfkVSzOdI ← ㅋㅋㅋ 진짜 안 넣고 싶었는데 각 스토리지 엔진의 특징 간단히 훑어보기 좋음
